main.cpp

```diff
@@ -4594,6 +4594,34 @@ void sign_guts_elf(elf_file* elf, private_t private_key, public_t public_key) {
         new_block.items.push_back(version);
     }
 
+    // Add entry point when signing Arm images
+    std::shared_ptr<image_type_item> image_type = new_block.get_item<image_type_item>();
+    if (settings.seal.sign && image_type != nullptr && image_type->image_type() == type_exe && image_type->cpu() == cpu_arm) {
+        std::shared_ptr<entry_point_item> entry_point = new_block.get_item<entry_point_item>();
+        if (entry_point == nullptr) {
+            std::shared_ptr<vector_table_item> vtor = new_block.get_item<vector_table_item>();
+            uint32_t vtor_loc = elf->header().entry < SRAM_START ? 0x10000000 : 0x20000000;
```

Comment:
do we need to check for XIP_SRAM-only binaries here? I forget if they can be signed (i believe so)

Reply:
I've added support for this - although I'm not sure if we support XIP_SRAM binaries without a vector_table_item?